### PR TITLE
[SYCL][UR][L0] Correct default for SYCL_PI_LEVEL_ZERO_FILTER_EVENT_WAIT_LIST

### PR DESCRIPTION
**Pull request by jandres742**
As previously committed in:

https://github.com/intel/llvm/commit/dbde93f8eb96d6c7b29e2386aac7a7cd64aef84d

### CODE REVIEW

**Comment by jandres742** [comment]
hip is failing with following error. Changes are specific to L0 so this is not related:

```sh
Prepare all required actions
Getting action download info
Run ./devops/actions/cached_checkout
Run mkdir -p /__w/repo_cache_e[2](https://github.com/intel/llvm/actions/runs/5538026732/jobs/10107710113?pr=10319#step:6:2)e//intel/llvm
fatal: detected dubious ownership in repository at '/__w/repo_cache_e2e/intel/llvm'
To add an exception for this directory, call:

	git config --global --add safe.directory /__w/repo_cache_e2e/intel/llvm
Error: Process completed with exit code [12](https://github.com/intel/llvm/actions/runs/5538026732/jobs/10107710113?pr=10319#step:6:13)8.
```

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/event.cpp [hunk 1]
```diff
@@ -1094,24 +1094,29 @@ ur_result_t _ur_ze_event_list_t::createAndRetainUrZeEventList(
...
       for (uint32_t I = 0; I < EventListLength; I++) {
+        auto Queue = EventList[I]->UrQueue;
         {
           std::shared_lock<ur_shared_mutex> Lock(EventList[I]->Mutex);
           if (EventList[I]->Completed)
             continue;
 
-          // Poll of the host-visible events.
-          auto HostVisibleEvent = EventList[I]->HostVisibleEvent;
-          if (FilterEventWaitList && HostVisibleEvent) {
```
Comment (smaslov-intel) [inline sycl/plugins/unified_runtime/ur/adapters/level_zero/event.cpp:1104]: Hmm. I made the default setting to not do any filtering in the plugin in https://github.com/intel/llvm/commit/dbde93f8eb96d6c7b29e2386aac7a7cd64aef84d

This somehow got lost with the move to adapter.
With it I think this change is not necessary.

Comment (jandres742) [reply sycl/plugins/unified_runtime/ur/adapters/level_zero/event.cpp:1104]: Thanks @smaslov-intel . Yes, it seems I missed this change https://github.com/intel/llvm/commit/dbde93f8eb96d6c7b29e2386aac7a7cd64aef84d

```cpp
const bool RetVal = Ret ? std::stoi(Ret) : 0;
```

I will do some testing on this.

**Comment by jandres742** [comment]
@smaslov-intel : change PR to add the default value as you did in https://github.com/intel/llvm/commit/dbde93f8eb96d6c7b29e2386aac7a7cd64aef84d

**Review by smaslov-intel — APPROVED** [review]

**Comment by jandres742** [comment]
Errors in hip unrelated as this touches only L0.

@intel/llvm-gatekeepers : please merge when possible.

**Pull request merged**